File: PreferencesSample/app/src/main/java/com/example/preferencessample/MyApplication.java
package com.example.preferencessample;

import android.app.Application;
import android.util.Log;

import androidx.room.Room;

public class MyApplication extends Application {

    public static MyApplication instance;
    private AppDatabase database;

    @Override
    public void onCreate() {
        super.onCreate();
        Log.e("MyApplication", "onCreate");
        instance = this;
        database = Room.databaseBuilder(this, AppDatabase.class, "contact_db")
                .fallbackToDestructiveMigration()
                .allowMainThreadQueries()
                .build();
    }

    public static MyApplication getInstance() {
        return instance;
    }

    public AppDatabase getDatabase() {
        return database;
    }

}

File: RoomSample/app/src/main/java/com/example/roomsample/MyApplication.kt
package com.example.roomsample

import android.app.Application
import android.util.Log
import androidx.room.Room


class MyApplication : Application() {
    private var database: AppDatabase? = null

    override fun onCreate() {
        super.onCreate()
        Log.e("MyApplication", "onCreate")
        instance = this
        database = Room.databaseBuilder(this, AppDatabase::class.java, "user_db")
            .fallbackToDestructiveMigration()
            .allowMainThreadQueries()
            .build()
    }

    fun getDatabase(): AppDatabase? {
        return database
    }

    companion object {
        lateinit var instance: MyApplication
    }

}